frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/OptionListEditor.tsx

```diff
@@ -48,137 +51,28 @@ export function OptionListEditor({
     case 'success': {
       if (optionsLists[optionsId] !== undefined) {
         return (
-          <EditLibraryOptionListEditorModal
-            label={label}
+          <LibraryOptionsEditor
+            component={component}
             optionsId={optionsId}
             optionsList={optionsLists[optionsId]}
+            handleDelete={handleDelete}
           />
         );
-      }
-      if (component.options !== undefined) {
+      } else if (component.options !== undefined) {
         return (
-          <EditManualOptionListEditorModal
-            label={label}
+          <ManualOptionsEditor
             component={component}
             handleComponentChange={handleComponentChange}
+            handleDelete={handleDelete}
           />
         );
+      } else {
```

Comment:
All these conditions should not be in the `success` case from the optionListQuery as only the `LibraryOptionEditor` is depending on it.